### ISSUE TITLE
Asynchronous initializers not working with externals

### ISSUE THREAD

**Issue by Prior99** [KnisterPeter/tsdi#1073]
Externals are not waiting for asynchronous dependencies when using `@initialize`.

```typescript
      @component
      class DeepNestedAsyncDependency {
        public value?: number;

        @initialize
        protected async init(): Promise<void> {
            // ...
        }
      }

      @component
      class SyncDependency {
        @inject public dependency!: DeepNestedAsyncDependency;
      }

      @external
      class Dependent {
        @inject public dependency!: SyncDependency;

        @initialize
        protected init(): void {
          assert.equal(this.dependency.dependency.value, 10);
          done();
        }
      }

      new Dependent();
```

**Comment by KnisterPeter** [comment]
Implemented in #1085

**Comment by KnisterPeter** [comment]
@shipjs prepare
